crates/bevy_reflect/src/list.rs

```diff
@@ -46,11 +46,11 @@ impl DynamicList {
 
 impl List for DynamicList {
     fn get(&self, index: usize) -> Option<&dyn Reflect> {
-        self.values.get(index).map(|value| &**value)
```

Comment:
Interesting; why were these changes required? 

They certainly seem reasonable, but I'm not familiar enough with the original code to really grasp why it was needed in the first place.

Reply:
It looks like we have reasonable (if basic) tests for this type, which is reassuring, as they continue to pass :)

Reply:
I think this is why it was done like this previously:
`get` called on `vec` results in an `Option<&Box<dyn Reflect>>`, in order to return an `Option<&dyn Reflect>` it uses `&**` (dereferences it twice to get to the trait object and then gets a reference to it).

Now, because of the new trait:
`Option<&Box<dyn Reflect>>`, the return value from `get`, already acts as `Option<&dyn Reflect>` (the return value), which already works.
The error I got was that `dyn Reflect` can't be dereferenced, so I don't know why it's acting as a `dyn Reflect` and not a `Box`, which is the actual type.
I believe this should work anyway since the Box will forward all reflect calls.

Reply:
> The error I got was that dyn Reflect can't be dereferenced, so I don't know why it's acting as a dyn Reflect and not a Box, which is the actual type.

Presumably it was dispatching directly to the `Box` type now that it implements reflect?

Awesome, your explanation makes sense. This is a nice improvement to the readability of that code, so I'm happy with it.

Reply:
I believe this should work, however I'm pretty sure this will result in double dynamic dispatch: the first dynamic call will call `Box<dyn Reflect>`'s method, which in turn will do a dynamic call on its inner value. Maybe this should need an `.as_deref()`?